FueledUtils/UIExtensions.swift

```diff
@@ -114,68 +150,190 @@ public extension UIView {
+	///
+	/// Refer to the documentation for `withAlpha(_:)` for more info.
+	///
+	@available(*, deprecated, renamed: "withAlpha(_:)")
 	public func imageWithAlpha(_ alpha: CGFloat) -> UIImage {
+		return self.withAlpha(alpha)
+	}
+
+	///
+	/// Returns the receiver with the given alpha applied to it. The rendering mode is kept the same.
+	/// The resulting image will be made resizable whether it was originally or not,
+	/// with `capInsets` set as the cap insets, and `resizingMode` as its resizing mode.
+	///
+	/// - Parameters:
+	///   - alpha: The alpha to apply to the receiver.
+	///
+	public func withAlpha(_ alpha: CGFloat) -> UIImage {
 		let rect = CGRect(origin: .zero, size: self.size)
 		let image = UIImage.draw(size: self.size, opaque: false, scale: self.scale) { _ in
 			self.draw(in: rect, blendMode: .normal, alpha: alpha)
 		}
 		return image.resizableImage(withCapInsets: self.capInsets, resizingMode: self.resizingMode).withRenderingMode(self.renderingMode)
 	}
 
+	///
+	/// **Deprecated**: Please use `withTint(_:)` instead.
+	///
+	/// Refer to the documentation for `withTint(_:)` for more info.
+	///
+	@available(*, deprecated, renamed: "withTint(_:)")
 	public func imageTintedWithColor(_ color: UIColor) -> UIImage {
+		return self.withTint(color)
+	}
+
+	///
+	/// Apply the given tint to the image. The rendering mode is kept the same.
+	/// The resulting image will be made resizable whether it was originally or not,
+	/// with `capInsets` set as the cap insets, and `resizingMode` as its resizing mode.
+	///
+	/// - Parameters:
+	///   - color: The color to apply to the receiver.
+	///
+	public func withTint(_ tint: UIColor) -> UIImage {
 		let image = UIImage.draw(size: self.size, scale: self.scale) { _ in
-			color.setFill()
+			tint.setFill()
 			UIRectFill(CGRect(origin: .zero, size: self.size))
 			self.draw(at: .zero, blendMode: .destinationIn, alpha: 1)
 		}
 		return image.resizableImage(withCapInsets: self.capInsets, resizingMode: self.resizingMode).withRenderingMode(self.renderingMode)
 	}
 
-	public func resized(offset: CGPoint = .zero, size: CGSize) -> UIImage? {
-		return UIImage.draw(size: size) { _ in
+	///
+	/// Apply the given tint to the image. The rendering mode is kept the same.
+	///
+	/// - Parameters:
+	///   - offset: The offset to apply to the original image in the resulting image.
+	///   - contextSize: The size of the context used to resize the image. If not specified, it defaults to `CGSize(width: offset.x + size.width, height: offset.y + size.height)`
+	///   - size: The size to resize the image to.
+	///
+	public func resized(offset: CGPoint = .zero, contextSize: CGSize? = nil, size: CGSize) -> UIImage? {
+		let contextSize = contextSize ?? CGSize(width: offset.x + size.width, height: offset.y + size.height)
+		return UIImage.draw(size: contextSize) { _ in
 			self.draw(in: CGRect(origin: offset, size: size))
-		}
+		}.withRenderingMode(self.renderingMode)
```

Comment:
Breaks backward compatibility, but for consistency with the others I've added it